gnmi_server/server_test.go

```diff
@@ -3610,11 +3610,202 @@ func TestParseOrigin(t *testing.T) {
 	}
 }
 
+func TestMasterArbitration(t *testing.T) {
+	s := createServer(t, 8088)
+	// Turn on Master Arbitration
+	s.ReqFromMaster = ReqFromMasterEnabledMA
+	go runServer(t, s)
+	defer s.s.Stop()
+
+	tlsConfig := &tls.Config{InsecureSkipVerify: true}
+	opts := []grpc.DialOption{grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig))}
+
+	//targetAddr := "30.57.185.38:8080"
+	targetAddr := "127.0.0.1:8088"
+	conn, err := grpc.Dial(targetAddr, opts...)
+	if err != nil {
+		t.Fatalf("Dialing to %q failed: %v", targetAddr, err)
+	}
+	defer conn.Close()
+
+	gClient := pb.NewGNMIClient(conn)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	maExt0 := &ext_pb.Extension{
+		Ext: &ext_pb.Extension_MasterArbitration{
+			MasterArbitration: &ext_pb.MasterArbitration{
+				ElectionId: &ext_pb.Uint128{High: 0, Low: 0},
+			},
+		},
+	}
+	maExt1 := &ext_pb.Extension{
+		Ext: &ext_pb.Extension_MasterArbitration{
+			MasterArbitration: &ext_pb.MasterArbitration{
+				ElectionId: &ext_pb.Uint128{High: 0, Low: 1},
+			},
+		},
+	}
+	maExt1H0L := &ext_pb.Extension{
+		Ext: &ext_pb.Extension_MasterArbitration{
+			MasterArbitration: &ext_pb.MasterArbitration{
+				ElectionId: &ext_pb.Uint128{High: 1, Low: 0},
+			},
+		},
+	}
+	regExt := &ext_pb.Extension{
+		Ext: &ext_pb.Extension_RegisteredExt{
+			RegisteredExt: &ext_pb.RegisteredExtension{},
+		},
+	}
+
+	// By default ElectionID starts from 0 so this test does not change it.
+	t.Run("MasterArbitrationOnElectionIdZero", func(t *testing.T) {
+		req := &pb.SetRequest{
+			Prefix: &pb.Path{Elem: []*pb.PathElem{{Name: "interfaces"}}},
+			Update: []*pb.Update{
+				newPbUpdate("interface[name=Ethernet0]/config/mtu", `{"mtu": 9104}`),
+			},
+			Extension: []*ext_pb.Extension{maExt0},
+		}
+		_, err = gClient.Set(ctx, req)
+		if err != nil {
+			t.Fatal("Did not expected an error: " + err.Error())
+		}
+		if _, ok := status.FromError(err); !ok {
```

Comment:
This condition can be removed. It will always be false due to the nil check above.

Reply:
I think we should be good. 
`If err is nil, a Status is returned with codes.OK and no message.`
Ref: https://pkg.go.dev/google.golang.org/grpc/status#FromError